### ISSUE TITLE
Gazelle doesn't honor goproxy

### ISSUE THREAD

**Issue by VivekSubr** [bazelbuild/bazel-gazelle#1328]
### What version of gazelle are you using?

0.22.2

### What version of rules_go are you using?

0.27.0

### What version of Bazel are you using?

4.1

### What operating system and processor architecture are you using?

Ubuntu x86, 20.04

### What did you do?

Set GOPROXY to a CI (artifactory) cache mirror, and expected go to be configured to query that url.

Project is envoy, here's the config: https://github.com/envoyproxy/envoy/blob/main/bazel/dependency_imports.bzl

### What did you expect to see?
This works expect for gazelle, which doesn't seem to be honoring GOPROXY. I expected gazelle to query the goproxy instead of https://golang.org/

### What did you see instead?
2022-08-29T15:58:27.5316804Z   /build/envoy-src/bazel/dependency_imports.bzl:19:27: in envoy_dependency_imports
2022-08-29T15:58:27.5318154Z   /***/.cache/bazel/_bazel_***/e08c0ad6c4a0c20d4308e47a29f89ff1/external/io_bazel_rules_go/go/private/sdk.bzl:453:28: in go_register_toolchains
2022-08-29T15:58:27.5319222Z   /***/.cache/bazel/_bazel_***/e08c0ad6c4a0c20d4308e47a29f89ff1/external/io_bazel_rules_go/go/private/sdk.bzl:129:21: in go_download_sdk
2022-08-29T15:58:27.5319973Z Repository rule _go_download_sdk defined at:
2022-08-29T15:58:27.5334711Z   /***/.cache/bazel/_bazel_***/e08c0ad6c4a0c20d4308e47a29f89ff1/external/io_bazel_rules_go/go/private/sdk.bzl:116:35: in <toplevel>
2022-08-29T15:58:27.5381842Z [0m[91m[35mWARNING: [0mDownload from https://golang.org/dl/?mode=json&include=all failed: class javax.net.ssl.SSLHandshakeException Remote host terminated the handshake

**Comment by VivekSubr** [comment]
Looks like bazel rules_go queries golang.org if sdks aren't specified... https://github.com/bazelbuild/rules_go/blob/master/go/private/sdk.bzl

Perhaps we are forced to specify sdks manually using [go_download_sdk](https://github.com/bazelbuild/rules_go/blob/master/go/toolchains.rst#go_download_sdk)?

**Comment by ttpathan** [comment]
Yeah, the error you reported is because of rules_go, not gazelle. If you manually specify sdks then it will work
```py
go_download_sdk(
    name="go_sdk",
    sdks = {
        "linux_amd64": ("go1.18.2.linux-amd64.tar.gz", "e54bec97a1a5d230fc2f9ad0880fcbabb5888f30ed9666eca4a91c5a32e86cbc"),
        "darwin_amd64": ("go1.18.2.darwin-amd64.tar.gz", "1f5f539ce0baa8b65f196ee219abf73a7d9cf558ba9128cc0fe4833da18b04f2"),
        "darwin_arm64": ("go1.18.2.darwin-arm64.tar.gz", "6c7df9a2405f09aa9bab55c93c9c4ce41d3e58127d626bc1825ba5d0a0045d5c"),
    },
)
```

For gazelle, we pass through GOPROXY as repo_env, not sure if there is a better way. Would be great if there was a way to define it in workspace. `.bazelrc`:
```
common --repo_env GOPROXY="https://url/to/artifactory"
common --repo_env GOPRIVATE="mycompany.com"
```

**Comment by tanner0101** [comment]
Add this to .bazelrc seems to work for me. Strange that this is the only Bazel rules that doesn't seem to obey url rewrite config.

```
build --action_env GOPROXY=https://art.company.corp/artifactory/go-proxy-golang
```

**Comment by hunshcn** [comment]
> Adding this to .bazelrc seems to work for me. Strange that this is the only Bazel rules that doesn't seem to obey url rewrite config.
> 
> ```
> build --action_env GOPROXY=https://art.company.corp/artifactory/go-proxy-golang
> ```

Yes, it do not use bazel downloader now.
